docs/10-architecture.md

```diff
@@ -0,0 +1,135 @@
+# Architecture
+
+## Graph
+
+```mermaid
+graph LR
```

Comment:
I spent a while trying to figure out whether we could make the diagram nodes link to the things (e.g. codebases) that they’re talking about, but unfortunately it looks like [markdown links are not supported by mermaid](https://github.com/mermaid-js/mermaid/issues/2239), and attempting to use a raw html link didn’t work in the github output either (it made a blue underline but didn’t work as a hyperlink). Trying to use mermaid locally to generate an SVG also didn’t work for me (couldn’t get mermaid installed and working). It’s a shame, I think it would’ve been nice. (I mention here only in case someone else likes the idea and manages to get it working!)